src/main/java/org/folio/rest/impl/UsersAPI.java

```diff
@@ -640,6 +642,49 @@ public void getUsersProfilePictureByProfileId(String profileId, Map<String, Stri
       });
   }
 
+  @Override
+  @Stream
+  public void putUsersProfilePictureByProfileId(String profileId, InputStream entity, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    try (InputStream bis = new BufferedInputStream(entity)) {
+      if (Objects.isNull(okapiHeaders.get(STREAM_ABORT))) {
+        validateAndProcessByteArray(bis);
+        if (Objects.nonNull(requestBytesArray) && requestBytesArray.length != 0) {
```

Comment:
I think this method can be split to two parts one is to check the file type and then rest to process when file type validation succeeds.